app/code/Magento/Widget/Controller/Adminhtml/Widget/Instance/Validate.php

```diff
@@ -19,6 +19,9 @@ public function execute()
         $response->setError(false);
         $widgetInstance = $this->_initWidgetInstance();
         $result = $widgetInstance->validate();
+        if ($result instanceof \Magento\Framework\Phrase) {
+            $result = (string) $result;
+        }
         if ($result !== true && is_string($result)) {
```

Comment:
I suggest changing this into:

```
if ($result !== true && $result instanceof \Magento\Framework\Phrase) {
    $this->messageManager->addError((string)$result);
    // ...
}
```

This way we can avoid the first `if...` and the result would be cleaner.

Let me know, thank you!

Reply:
It still needs to check if $result is a string, so I was trying to avoid putting an AND and an OR on the same line by isolating the Phrase evaluation. However, I've reformatted it as you suggested.